aten/src/ATen/native/cpu/PowKernel.cpp

```diff
@@ -35,66 +37,115 @@ void pow_tensor_tensor_kernel(TensorIterator& iter) {
 }
 
 void pow_tensor_scalar_kernel(TensorIterator& iter, Scalar exp_scalar) {
-  if (isFloatingType(iter.dtype())) {
+  // prevent multiple calls to iter.common_dtype()
+  const auto dtype = iter.common_dtype();
+  if (isFloatingType(dtype)) {
```

Comment:
This looks much better. What about a small revision of the conditional structure to be:

```
if (float or double) {
} else if (half) {
} else if (bfloat16) {
} else {
}
```

I think that would simplify things even further?

Reply:
Yes, since it turns out that the general case can't combine `BFloat16` & `Float16` with `float` & `double`, I had also thought of reverting the conditional structure to this one.